app/ts/simulation/services/SimulationModeEthereumClientService.ts

```diff
@@ -918,3 +918,27 @@ export const appendSignedMessage = async (ethereumClientService: EthereumClientS
 		signedMessages: simulationState.signedMessages.concat(signedMessage)
 	}
 }
+
+// takes the most recent block that the application is querying and does the calculation based on that
+export const getSimulatedFeeHistory = async (ethereumClientService: EthereumClientService, simulationState: SimulationState | undefined, request: FeeHistory): Promise<EthGetFeeHistoryResponse> => {
+	//const numberOfBlocks = request.params[0]
+	const blockTag = request.params[1]
+	const rewardPercentiles = request.params[2]
+	const newestBlock = await getSimulatedBlock(ethereumClientService, simulationState, blockTag, true)
+	const blocksToConsider = [newestBlock]
+	const oldestBlock = blocksToConsider[0]?.number
+	if (oldestBlock === undefined) throw new Error('oldest block was undefined')
+	return {
+		baseFeePerGas: blocksToConsider.map((block) => block.baseFeePerGas).filter((baseFeePerGas): baseFeePerGas is bigint => !!baseFeePerGas),
+		gasUsedRatio: blocksToConsider.map((block) => Number(block.gasUsed) / Number(block.gasLimit)).filter((gasUsedRatio): gasUsedRatio is number => !!gasUsedRatio),
+		oldestBlock,
+		...rewardPercentiles === undefined ? {} : {
+			reward: blocksToConsider.map((block) => rewardPercentiles.map((percentile) => {
+				const effectivePriority = block.transactions.map((tx) => tx.type === '1559' ? min(tx.maxPriorityFeePerGas, tx.maxFeePerGas - (block.baseFeePerGas ?? 0n)) : tx.gasPrice - (block.baseFeePerGas ?? 0n))
+				// we are using transaction.gas as a weighting factor while this should be gas used. Getting gas used requires getting receipts which is difficult
+				const gasWeights = block.transactions.map((tx) => tx.gas)
+				return calculateWeightedPercentile(effectivePriority, gasWeights, BigInt(percentile))
```

Comment:
Recommend pluralizing.  I spent a while confused because I thought this was a single value, rather than a collection.
```suggestion
				const effectivePriorityFees = block.transactions.map((tx) => tx.type === '1559' ? min(tx.maxPriorityFeePerGas, tx.maxFeePerGas - (block.baseFeePerGas ?? 0n)) : tx.gasPrice - (block.baseFeePerGas ?? 0n))
				// we are using transaction.gas as a weighting factor while this should be gas used. Getting gas used requires getting receipts which is difficult
				const gasWeights = block.transactions.map((tx) => tx.gas)
				return calculateWeightedPercentile(effectivePriorityFees, gasWeights, BigInt(percentile))
```

Reply:
It feels like `calculateWeightedPercentile` should take an array of tuples or objects like:
```ts
[
	[effectivePriorityFee1, gasLimit1],
	[effectivePriorityFee2, gasLimit2],
]
// or
[
	{ effectivePriorityFee: effectivePriorityFee1, gasLimit: gasLimit1 },
	{ effectivePriorityFee: effectivePriorityFee2, gasLimit: gasLimit2 },
]
```
Passing in two separate arrays that *should* line up is generally a bad pattern as it can lead to easy mistakes the compiler cannot protect against.